NodeJS HW2/index.js

```diff
@@ -0,0 +1,136 @@
+const { v4: uuidv4 } = require("uuid");
+const express = require("express");
+const Joi = require("joi");
+
+const app = express();
+const router = express.Router();
+app.use(express.json());
+
+const userSchema = Joi.object().keys({
+  login: Joi.string().min(2).required(),
+  password: Joi.string()
+    .regex(/^[a-zA-Z0-9]{3,30}$/)
+    .required(),
+  age: Joi.number().integer().min(4).max(130),
+});
+
+const errorResponse = (schemaErrors) => {
+  const errors = schemaErrors.map(({ path, message }) => ({
+    path,
+    message,
+  }));
+
+  return {
+    status: "failed",
+    errors,
+  };
+};
+
+const validateSchema = (schema) => {
+  return (req, res, next) => {
+    const { error } = schema.validate(req.body, {
+      abortEarly: false,
+      allowUnknown: false,
+    });
+    if (error?.isJoi) {
+      res.status(400).json(errorResponse(error.details));
+    } else {
+      next();
+    }
+  };
+};
+
+const users = [
+  {
+    id: uuidv4(),
+    login: "login1",
+    password: "abc123",
+    age: 20,
+    isDeleted: false,
+  },
+  {
+    id: uuidv4(),
+    login: "login2",
+    password: "abc345",
+    age: 25,
+    isDeleted: false,
+  },
+  {
+    id: uuidv4(),
+    login: "login3",
+    password: "abc567",
+    age: 30,
+    isDeleted: false,
+  },
+];
+
+const getExistingUsers = () => {
+  return users.filter((user) => !user.isDeleted);
+};
+
+router.param("id", (req, res, next, id) => {
+  const user = users.find((user) => user.id === id);
+  if (!user)
+    res.status(404).json({ message: `User with id=${id} was not found!` });
+  else {
+    res.locals.user = user;
+    next();
+  }
+});
+
+router.get("/", (req, res) => {
+  const users = getExistingUsers();
+  const { limit, loginSubstring } = req.body;
```

Comment:
it's not a good practice to include body in GET request, plz consider other alternatives

Reply:
I will use query params then.